pyproject.toml

```diff
@@ -31,20 +33,87 @@ license={text="MIT"}
 #   Pygraphviz is often tricky to install, so we reserve it for the dev extras
 #   list.
 # - six: Required by auto-generated Swagger models
-dev = ["fastapi", "requests", "uvicorn", "pytest", "black", "pygraphviz", "six"]
+dev = ["fastapi", "requests", "uvicorn", "pytest", "pytest-cov", "pytest-xdist", "black", "mypy", "coverage", "pygraphviz", "six"]
 
 # MOVIZ dependencies. MOVIZ is a webapp for visualizing function networks (see
 # the skema/moviz directory).
 moviz = ["flask", "graphviz"]
 
+# ISA dependencies.
+isa = [
+  "pydot", 
+  "graspologic" # NOTE: this depends on rust
+]
+
 # Im2MML dependencies. The img2mml service converts equation images to MathML.
 # See the skema/img2mml directory.
-img2mml = ["fastapi", "requests", "uvicorn", "torch", "torchvision", "python-multipart"]
+img2mml = ["fastapi", "uvicorn", "torch", "torchvision", "python-multipart"]
+
+# dependencies for text reading utilities.
+tr = ["transformers", "gensim", "SPARQLWrapper", "tqdm", "pandas"]
+
+# project documentation generation
+doc = ["mkdocs==1.2.3", "pdoc3==0.10.0", "mkdocs-git-snippet==0.1.1", "mkdocs-git-revision-date-localized-plugin==0.11.1", "mkdocs-git-authors-plugin==0.6.3", "mkdocs-rtd-dropdown==1.0.2", "jinja2<3.1.0"]
+
+core = ["skema[moviz]", "skema[img2mml]", "skema[isa]", "skema[tr]"]
+
+# all extras
+all = ["skema[core]", "skema[dev]", "skema[doc]"]
+
+[tool.setuptools.package-dir]
+"skema.gromet" = "skema/gromet"
+"skema.img2mml" = "skema/img2mml"
+"skema.isa" = "skema/isa"
+"skema.moviz" = "skema/moviz"
+"skema.model_assembly" = "skema/model_assembly"
+"skema.program_analysis" = "skema/program_analysis"
+"skema.skema_py" = "skema/skema_py"
+# re-map skema/text_reading/python to skema.text_reading
+"skema.text_reading" = "skema/text_reading/python"
```

Comment:
@myedibleenso : Let's keep in mind that the python code that lives under `mention_linking` will be deprecated in favor of `model_linker` introduced in #246

Reply:
@enoriega , if you want to open a PR for this branch that replaces that code, feel free to pull these changes into another branch and do so.  Otherwise we can take care of it in a subsequent PR.